src/persist/src/operators/source.rs

```diff
@@ -99,24 +94,31 @@ where
             }
         }
 
-        let mut cap = Some(capability);
+        // TODO: This currently works by only emitting the persisted data on worker
+        // 0 because that was the simplest thing to do initially. Instead, we should
+        // shard up the responsibility between all the workers.
+        let (mut cap, mut done) = match (listen_rx.as_ref(), worker_index) {
+            (Some(_), 0) => (Some(capability), false),
```

Comment:
this logic seems non-obvious to me. without thinking too hard about it, it seems cleaner if we have a single Option<(Capability, Receiver)> that we can drop?

Reply:
great idea! done